src/sentry/rules/actions/notify_event_sentry_app.py

```diff
@@ -0,0 +1,68 @@
+"""
+Used for notifying a *specific* sentry app with a custom webhook payload (i.e. specified UI components)
+"""
+
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.sentry_app_component import SentryAppAlertRuleActionSerializer
+from sentry.coreapi import APIError
+from sentry.mediators import sentry_app_components
+from sentry.models import SentryApp, SentryAppComponent, SentryAppInstallation
+from sentry.rules.actions.base import EventAction
+from sentry.tasks.sentry_apps import notify_sentry_app
+
+
+class NotifyEventSentryAppAction(EventAction):
+    actionType = "sentryapp"
+    # Required field for EventAction, value is ignored
+    label = ""
+
+    # TODO(Leander): As there is no form_cls (e.g. NotifyEventSentryAppActionForm) the form data will
+    # not be validated on the backend. This is tricky to do since the schema form is dynamic, and will
+    # be implemented on it's own in the future. Frontend validation is still in place in the mean time.
+
+    def get_custom_actions(self, project):
+        action_list = []
+        for install in SentryAppInstallation.get_installed_for_org(project.organization_id):
+            _components = SentryAppComponent.objects.filter(
+                sentry_app_id=install.sentry_app_id, type="alert-rule-action"
+            )
+            for component in _components:
+                try:
+                    sentry_app_components.Preparer.run(
+                        component=component, install=install, project=project
+                    )
+                    # These details are unique to the Sentry App
+                    action_details = serialize(
+                        component, None, SentryAppAlertRuleActionSerializer(), install=install
+                    )
+                    # These details are shared among each Sentry App action (with UI)
+                    action_details["id"] = self.id
```

Comment:
I'd pass the `EventAction` as `self` to the serializer to add these fields to `action_details`. That way you can reuse this `get_custom_actions()` in `ProjectRulesConfigurationEndpoint`.

Reply:
Re-worked the serializer to add the extra fields to action_details. I'm unsure what you mean by reuse `get_custom_actions()` in `ProjectRulesConfigurationEndpoint`, as that is the primary place it is used. I think it's because I left the duplicate code in there, which has now been deleted. This code does not get used anywhere else.